torch/csrc/jit/serialization/unpickler.cpp

```diff
@@ -532,6 +532,21 @@ PickleOpCode Unpickler::readInstruction() {
       }
       stack_.emplace_back(std::move(tensor));
     } break;
+    case PickleOpCode::SETITEM: {
+      // At this OpCode, stack looks like
+      // | Stack Top |
+      // | ......    |
+      // | Dict      | -> (stack_size - 3)
+      // | Key       | -> (stack_size - 2)
+      // | Value     | -> (stack_size - 1)
+      auto stack_size = stack_.size();
+      auto dict_pos = stack_size - 3;
+      auto key_pos = stack_size - 2;
+      auto val_pos = stack_size - 1;
+      auto dict = stack_.at(dict_pos).toGenericDict();
+      dict.insert_or_assign(stack_.at(key_pos), stack_.at(val_pos));
+      stack_.erase(stack_.begin() + (key_pos), stack_.end());
```

Comment:
The top of the stack after this points to the Dict. That is expected right?

Reply:
Yes. 

(Also, SETITEMS does the same thing : https://github.com/pytorch/pytorch/blob/20fc474410f49a8b9707297ce778425541a9a131/torch/csrc/jit/serialization/unpickler.cpp#L433)